### PR TITLE
Implemented a $lock->clear method

### PR DESCRIPTION
**Pull request by remipelhate**
Basically, it does the same thing as `$lock->deny()`, except it doesn't add a restriction for it. Let me know if I'm forgetting something here.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,4 @@
 vendor/
 .DS_Store
-composer.lock
\ No newline at end of file
+composer.lock
+.idea
```
Comment (driesvints) [inline .gitignore]: Please ignore your IDE settings through a global gitignore

**Comment by remipelhate** [comment]
@driesvints I added some more tests, let me know if I'm still missing some. If you approve it, I'll squash my changes to one commit.

**Comment by driesvints** [comment]
@remipelhate can you add some more tests for the persistent driver test case as well? It's probably even more important there that everything works fine.

**Comment by remipelhate** [comment]
Sure, on it!

**Comment by driesvints** [comment]
Thanks @remipelhate !

**Pull request merged**